README.md

```diff
@@ -1,2 +1,101 @@
 # dart_multihash
-A dart-lang implementation of the Multihash protocol
+
+This is an implementation of the 
+[Multihash](https://github.com/multiformats/multihash)
+standard in Dart.
+This package provides an extensible representation of cryptographic hashes.
+
+# Install
+
+To install `dart_multihash`, simply run the following command.
+
+```elixir
+flutter pub add dart_multihash
+```
+
+# Usage
+
+This package simply encodes and decodes
+hashes with the Multihash standard. 
+This is a low-level library. 
+So bring your own digest.
+To create hashes, you will most likely
+need to use a package
+like [`crypto`](https://pub.dev/packages/crypto)
+to hash a given input.
+
+Here's the standard use of `dart_multihash`,
+using the `encode()` and `decode()` functions.
+
+```dart
+import 'package:crypto/crypto.dart';
```

Comment:
Why are people having to _manually_ import `package:crypto`? is this not a dependency of `Multihash`? 🤔 
If not, _should_ it be to streamline the usage? 💭

Reply:
Because multihashing just encodes whatever digest it receives with the given hash function and digest length. It just follows [the multiformats convention](https://github.com/multiformats/multicodec/blob/master/table.csv). 
Encrypting a hash is responsibility of who is creating [the `cid`](https://github.com/dwyl/dart_cid/pull/3) and encoding the multihash with whatever hash function they used to encrypt the hash. 

If you check any of the [multihash implementations](https://github.com/multiformats/multihash#implementations), they just encode digests following convention, not encrypt anything.